ynput/ayon-core
# Houdini: Sync publisher attributes with node parameters

## Changelog Description
Use Houdini node parameters as the source of truth while using metadata as temp values.
This PR adds two methods to creator class which sync the attributes in the publisher to Houdini nodes.
1. `update_node_parameters` update node parameters according the metadata (user preference in the publish tab.) _It's triggered when user click save button in publisher_
2. `read_node_data` update metadata according to the node parameters. _It's triggered when user click refresh button in publisher_

Things left: check these product types:
|Product Type|`update_node_parameters` | `read_node_data` | 
|----------------|---------------|---------------|
|Camera (Abc)| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|Arnold ASS| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|Arnold ROP| <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> |
|PointCache (Bgeo)| <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> |
|Composite (Image Sequence)| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|Houdini Digital Asset (Hda)| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|Karma ROP| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|Mantra IFD| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|Mantra ROP| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|PointCache (Abc)| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|Redshift Proxy| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|Redshift ROP| <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> |
|Review| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|Static Mesh (FBX) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|VDB Cache| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|VRay ROP| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
|USD (experimental)| <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |

## Demo
https://github.com/ynput/ayon-core/assets/20871534/fbeaff1f-6c1c-494a-8bc2-b0b8af57db01


## Testing notes:
1. Open Houdini
2. Create instances 
3. In publish tab, change some options and hit save button (you should find out that node parameters are synced with the selected options in the publisher)
4. Change the dedicated nodes parameters and hit refresh (you should find out the publisher is synced to the node)
